vector-im/element-x-android
# Integrate mentions in the composer

## Type of change


## Content

- Integrates a new version of the RTE library with better support for mentions.
- Sets a callback in the mention suggestion selection view to actually insert the selection.
- Creates `MentionSpanProvider` to provide the RTE composer with the custom `ReplacementSpan` for mentions.

## Motivation and context

Closes #1453 

## Screenshots / GIFs

|Light|Dark|
|-|-|
|<img width="567" alt="image" src="https://github.com/vector-im/element-x-android/assets/480955/c3718506-3db2-43aa-b0ed-4776fe1f0ba2">|<img width="565" alt="image" src="https://github.com/vector-im/element-x-android/assets/480955/d18671cd-e3c7-49ea-a541-3f35c4d76938">|

## Tests



- Open a room, type `@` in the composer.
- Select something from the list of suggestions.
- A mention pill should be added to the editor and when sending, it'll be recognised as an actual mention.

## Tested devices

- OS version(s): 11

## Checklist
